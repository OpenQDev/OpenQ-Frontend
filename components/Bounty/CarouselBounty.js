import React, { useContext, useMemo } from 'react';
import StoreContext from '../../store/Store/StoreContext';
import useGetTokenValues from '../../hooks/useGetTokenValues';
import Image from 'next/image';
import ToolTipNew from '../Utils/ToolTipNew';
import Link from 'next/link';
import { PersonAddIcon, PersonIcon, PeopleIcon } from '@primer/octicons-react';
import LabelsList from '../Bounty/LabelsList';
import useWeb3 from '../../hooks/useWeb3';

const CarouselBounty = ({ bounty }) => {
  const [appState] = useContext(StoreContext);
  const { safe } = useWeb3();
  const createBudget = (bounty) => {
    return bounty.fundingGoalTokenAddress
      ? {
          tokenAddress: bounty.fundingGoalTokenAddress,
          volume: bounty.fundingGoalVolume,
        }
      : null;
  };
  const budgetObj = useMemo(() => createBudget(bounty), [bounty]);
  const [budgetValue] = useGetTokenValues(budgetObj);
  const budget = budgetValue?.total;

  const [tokenValues] = useGetTokenValues(bounty?.bountyTokenBalances);
  const [payoutValues] = useGetTokenValues(bounty.payouts);
  const price = tokenValues?.total;

  return (
    <>
      <Link
        target={safe ? '_self' : '_blank'}
        rel='noopener noreferrer'
        href={`${process.env.NEXT_PUBLIC_BASE_URL}/contract/${bounty.bountyId}/${bounty.bountyAddress}`}
      >
        <a className='border-web-gray bg-dark-mode p-4 gap-2 border rounded-sm flex mb-1'>
          <div className='w-64'>
            <div className='flex flex-row justify-between items-center'>
              <p className='text-normal truncate'>
                {bounty.owner.toLowerCase()}/{bounty.repoName.toLowerCase()}
              </p>
              <div className='pt-1'>
                <svg
                  xmlns='http://www.w3.org/2000/svg'
                  fill={bounty?.STATUS === 'CLOSED' ? '#F0431D' : '#15FB31'}
                  viewBox='0 0 16 16'
                  width='19'
                  height='19'
                >
                  <path d='M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z'></path>
                  <path
                    fillRule='evenodd'
                    d='M8 0a8 8 0 100 16A8 8 0 008 0zM1.5 8a6.5 6.5 0 1113 0 6.5 6.5 0 01-13 0z'
                  ></path>
                </svg>
              </div>
            </div>
            <h4 className='font-semibold truncate leading-tight mb-1 pt-1'>{bounty.title.toLowerCase()}</h4>
            <div className='pt-1'>
              <LabelsList bounty={bounty} isLimited={true} />
            </div>
            <div className='pt-10'>
              <div className='flex flex-row justify-between'>
                <div></div>
                {bounty.bountyType === '0' ? (
                  <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                    <PersonIcon />
                    <div className='whitespace-nowrap'>Fixed Price</div>
                  </span>
                ) : bounty.bountyType === '1' ? (
                  <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                    <PersonAddIcon />
                    <div className='whitespace-nowrap'>Split Price</div>
                  </span>
                ) : (
                  (bounty.bountyType === '2' || bounty.bountyType === '3') && (
                    <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                      <PeopleIcon />
                      <div className='whitespace-nowrap'>Contest</div>
                    </span>
                  )
                )}
              </div>
              <div className='pt-1'>
                <div className='flex flex-row justify-between items-center'>
                  <div className='pt-1 items-center gap-4 text-muted font-semibold'>
                    <span>
                      <svg
                        xmlns='http://www.w3.org/2000/svg'
                        className={'stroke-muted inline-block mr-1 -mt-1 fill-muted'}
                        viewBox='0 0 16 16'
                        width='16'
                        height='16'
                      >
                        <path d='M1.679 7.932c.412-.621 1.242-1.75 2.366-2.717C5.175 4.242 6.527 3.5 8 3.5c1.473 0 2.824.742 3.955 1.715 1.124.967 1.954 2.096 2.366 2.717a.119.119 0 010 .136c-.412.621-1.242 1.75-2.366 2.717C10.825 11.758 9.473 12.5 8 12.5c-1.473 0-2.824-.742-3.955-1.715C2.92 9.818 2.09 8.69 1.679 8.068a.119.119 0 010-.136zM8 2c-1.981 0-3.67.992-4.933 2.078C1.797 5.169.88 6.423.43 7.1a1.619 1.619 0 000 1.798c.45.678 1.367 1.932 2.637 3.024C4.329 13.008 6.019 14 8 14c1.981 0 3.67-.992 4.933-2.078 1.27-1.091 2.187-2.345 2.637-3.023a1.619 1.619 0 000-1.798c-.45-.678-1.367-1.932-2.637-3.023C11.671 2.992 9.981 2 8 2zm0 8a2 2 0 100-4 2 2 0 000 4z'></path>
                      </svg>
                      <span>{bounty.watchingCount}</span>
                    </span>
                  </div>
                  <div className=''>
                    {price > budget ? (
                      <div className='flex flex-row space-x-1 items-center'>
                        <div className='pr-2 pt-1'>
                          <Image src='/crypto-logos/ETH-COLORED.png' alt='avatarUrl' width='12' height='20' />
                        </div>

                        {bounty.status !== '0' ? (
                          <>
                            <div className='font-semibold '>TVC</div>
                            <div className=''>
                              {appState.utils.formatter.format(bounty.tvc || payoutValues?.total || 0)}
                            </div>
                          </>
                        ) : (
                          <>
                            <div className='font-semibold '>TVL</div>
                            <div className=''>{appState.utils.formatter.format(price)}</div>
                          </>
                        )}
                      </div>
                    ) : budget > 0 ? (
                      <>
                        <div className='flex flex-row space-x-1 items-center'>
                          <div className='pr-2 pt-1'>
                            <Image src='/crypto-logos/ETH.svg' alt='avatarUrl' width='12' height='20' />
                          </div>

                          <>
                            <div className='font-semibold '>Budget</div>
                            <div className=''>{appState.utils.formatter.format(budget)}</div>
                          </>
                        </div>
                      </>
                    ) : (
                      <div className='flex gap-2'>
                        <div className='flex flex-row space-x-1 items-center'>
                          <div className='pr-2 pt-1'>
                            <Image src='/crypto-logos/ETH.svg' alt='avatarUrl' width='12' height='20' />
                          </div>

                          <>
                            <div className='font-semibold '>Budget</div>
                          </>
                        </div>
                        <div className='flex flex-row space-x-1 items-center'>
                          <ToolTipNew
                            innerStyles={'whitespace-normal w-60'}
                            toolTipText={'No budget has been set for this contract'}
                          >
                            <div className='cursor-help rounded-full border border-[#c9d1d9] aspect-square leading-4 h-4 box-content text-center font-bold text-primary'>
                              ?
                            </div>
                          </ToolTipNew>
                        </div>
                      </div>
                    )}
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
      </Link>
    </>
  );
};
export default CarouselBounty;
