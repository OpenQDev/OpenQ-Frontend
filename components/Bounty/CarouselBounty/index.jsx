import React, { useContext } from 'react';
import StoreContext from '../../../store/Store/StoreContext';
import Skeleton from 'react-loading-skeleton';
import Image from 'next/image';
import ToolTipNew from '../../Utils/ToolTipNew';
import Link from 'next/link';
import { PersonAddIcon, PersonIcon, PeopleIcon } from '@primer/octicons-react';
import LabelsList from '../LabelsList';
import useWeb3 from '../../../hooks/useWeb3';
import useDisplayValue from '../../../hooks/useDisplayValue';

const CarouselBounty = ({ bounty }) => {
  const [appState] = useContext(StoreContext);
  const { safe } = useWeb3();
  const displayValue = useDisplayValue(bounty, appState.utils.formatter.format);

  return (
    <div className='border-web-gray bg-dark-mode p-4 gap-2 border rounded-sm flex mb-1'>
      <Link
        target={safe ? '_self' : '_blank'}
        rel='noopener noreferrer'
        href={`${process.env.NEXT_PUBLIC_BASE_URL}/contract/${bounty.bountyId}/${bounty.bountyAddress}`}
        className='flex flex-col justify-between w-72'
      >
        <div>
          <div className='flex flex-row justify-between items-start'>
            <div className='flex flex-col justify-center items-start'>
              <p className='text-normal truncate'>
                {bounty.alternativeName ? <span className='whitespace-nowrap'>{bounty.alternativeName}</span> : ''}
              </p>
              <p className='text-normal truncate mr-1'>
                {bounty.owner}/{bounty.repoName}
              </p>
            </div>
            <div className='pt-1'>
              <svg
                xmlns='http://www.w3.org/2000/svg'
                className={bounty?.status === '1' ? 'fill-danger' : 'fill-green'}
                viewBox='0 0 16 16'
                width='19'
                height='19'
              >
                <path d='M8 9.5a1.5 1.5 0 100-3 1.5 1.5 0 000 3z'></path>
                <path
                  fillRule='evenodd'
                  d='M8 0a8 8 0 100 16A8 8 0 008 0zM1.5 8a6.5 6.5 0 1113 0 6.5 6.5 0 01-13 0z'
                ></path>
              </svg>
            </div>
          </div>
          <h4 className='font-semibold leading-tight mb-1 pt-1 h-12 overflow-y-auto'>{bounty.title}</h4>
        </div>
        <div className='pt-1'>
          <LabelsList bounty={bounty} isLimited={true} />
        </div>
        <div className='pt-2'>
          <div className='flex flex-row justify-between'>
            <div></div>
            {bounty.bountyType === '0' ? (
              <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                <PersonIcon />
                <div className='whitespace-nowrap'>Fixed Price</div>
              </span>
            ) : bounty.bountyType === '1' ? (
              <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                <PersonAddIcon />
                <div className='whitespace-nowrap'>Split Price</div>
              </span>
            ) : (
              (bounty.bountyType === '2' || bounty.bountyType === '3') && (
                <span className='font-semibold flex flex-end items-center content-center gap-1 w-max'>
                  <PeopleIcon />
                  <div className='whitespace-nowrap'>Contest</div>
                </span>
              )
            )}
          </div>
          <div className='pt-1'>
            <div className='flex flex-row justify-between items-center'>
              <div className='pt-1 items-center gap-4 text-muted font-semibold'>
                <span>
                  <svg
                    xmlns='http://www.w3.org/2000/svg'
                    className={'stroke-muted inline-block mr-1 -mt-1 fill-muted'}
                    viewBox='0 0 16 16'
                    width='16'
                    height='16'
                  >
                    <path d='M1.679 7.932c.412-.621 1.242-1.75 2.366-2.717C5.175 4.242 6.527 3.5 8 3.5c1.473 0 2.824.742 3.955 1.715 1.124.967 1.954 2.096 2.366 2.717a.119.119 0 010 .136c-.412.621-1.242 1.75-2.366 2.717C10.825 11.758 9.473 12.5 8 12.5c-1.473 0-2.824-.742-3.955-1.715C2.92 9.818 2.09 8.69 1.679 8.068a.119.119 0 010-.136zM8 2c-1.981 0-3.67.992-4.933 2.078C1.797 5.169.88 6.423.43 7.1a1.619 1.619 0 000 1.798c.45.678 1.367 1.932 2.637 3.024C4.329 13.008 6.019 14 8 14c1.981 0 3.67-.992 4.933-2.078 1.27-1.091 2.187-2.345 2.637-3.023a1.619 1.619 0 000-1.798c-.45-.678-1.367-1.932-2.637-3.023C11.671 2.992 9.981 2 8 2zm0 8a2 2 0 100-4 2 2 0 000 4z'></path>
                  </svg>
                  <span>{bounty.watchingCount}</span>
                </span>
              </div>
              <div className=''>
                {displayValue ? (
                  <div className='flex flex-row space-x-1 w-min items-center'>
                    <div className='pr-2 pt-1 w-4'>
                      <Image
                        src={displayValue?.imgSrc || '/crypto-logos/ETH.svg'}
                        alt='avatarUrl'
                        width='12'
                        height='20'
                      />
                    </div>
                    {displayValue.displayValue ? (
                      <>
                        <div className='font-semibold '>{displayValue?.valueType}</div>
                        <div className=''>{displayValue?.displayValue}</div>
                      </>
                    ) : (
                      <>
                        <div className='font-semibold '>Budget</div>
                        <div className='flex flex-row space-x-1 items-center'>
                          <ToolTipNew
                            innerStyles={'whitespace-normal w-60'}
                            toolTipText={'No budget has been set for this contract'}
                          >
                            <div className='cursor-help p-0.25 rounded-full border border-[#c9d1d9] aspect-square leading-4 h-4 box-content text-center font-bold text-primary'>
                              ?
                            </div>
                          </ToolTipNew>
                        </div>
                      </>
                    )}
                  </div>
                ) : (
                  <Skeleton width={100} baseColor='#333' borderRadius={'1rem'} />
                )}
              </div>
            </div>
          </div>
        </div>
      </Link>
    </div>
  );
};
export default CarouselBounty;
