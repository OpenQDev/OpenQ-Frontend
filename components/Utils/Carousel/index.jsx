// Third party
import React, { useEffect, useRef, useState } from 'react';

const Carousel = ({ styles, children, height }) => {
  const scroller = useRef();
  const [isScrollable, setIsScrollable] = useState();
  const [isScrolling, setIsScrolling] = useState();

  // Thanks to https://stackoverflow.com/questions/36862334/get-viewport-window-height-in-reactjs
  useEffect(() => {
    function updateScrollable() {
      setIsScrollable(scroller.current?.scrollLeftMax > 0);
    }
    window.addEventListener('resize', updateScrollable);
    setTimeout(updateScrollable, 300);
    return () => window.removeEventListener('resize', updateScrollable);
  }, []);

  useEffect(() => {
    let incrementer;
    if (isScrolling) {
      const scrollVector = isScrolling === 'right' ? 10 : -10;
      incrementer = setInterval(() => {
        const currentScroll = scroller.current.scrollLeft;
        scroller.current.scroll({
          left: scrollVector + currentScroll,
          behaviour: 'smooth',
        });
      }, 0.25);
    }
    return () => {
      if (isScrolling) {
        clearInterval(incrementer);
      }
    };
  }, [isScrolling]);

  // render
  return (
    <div
      onMouseUp={() => setIsScrolling()}
      className={`relative ${
        isScrollable && 'xl:-left-10 xl:-right-10  xl:w-[calc(100%+80px)]'
      } justify-center self-center w-full flex ${styles}`}
    >
      {isScrollable && (
        <button
          className='sr-only xl:not-sr-only'
          data-testid='scroll carousel left'
          onMouseDown={() => setIsScrolling('left')}
        >
          <svg
            className='hover:fill-web-gray fill-dark-mode stroke-web-gray'
            width='44'
            height='36'
            viewBox='0 0 44 66'
            xmlns='http://www.w3.org/2000/svg'
          >
            <path
              d='M36.0583 63.6563L3.17933 35.4362C1.29394 33.8179 1.32452 30.8904 3.24329 29.3119L36.0513 2.32171C39.4475 -0.472269 44.2094 3.46345 42.1048 7.32496L29.9165 29.688C28.9554 31.4514 28.9403 33.5786 29.8762 35.3555L42.2025 58.7569C44.259 62.6611 39.4068 66.5303 36.0583 63.6563Z'
              strokeWidth='2'
            />
          </svg>
        </button>
      )}
      <div
        ref={scroller}
        data-testid='carousel'
        className={`flex gap-2 xl:max-w-[100%] xl:w-[980px] ${
          isScrollable && 'pb-3'
        } overflow-x-auto justify-items-center`}
      >
        {isScrollable && (
          <div
            className={`pointer-events-none absolute bg-gradient-to-r from-dark-mode to-black/0 w-12 sm:w-52 h-${
              height || 36
            } left-0 xl:left-10`}
          ></div>
        )}
        {children}
        {isScrollable && (
          <div
            className={`absolute pointer-events-none bg-gradient-to-l from-dark-mode to-black/0 w-12 sm:w-52 h-${
              height || 36
            } right-0 xl:right-10`}
          ></div>
        )}
      </div>
      {isScrollable && (
        <button
          className='sr-only xl:not-sr-only'
          data-testid='scroll carousel right'
          onMouseDown={() => setIsScrolling('right')}
        >
          <svg
            className='hover:fill-web-gray fill-dark-mode stroke-web-gray'
            width='44'
            height='36'
            viewBox='0 0 44 66'
            xmlns='http://www.w3.org/2000/svg'
          >
            <path
              d='M8.56771 63.6563L41.4466 35.4362C43.332 33.8179 43.3015 30.8904 41.3827 29.3119L8.57468 2.32171C5.17844 -0.472269 0.416603 3.46345 2.5212 7.32496L14.7095 29.688C15.6706 31.4514 15.6857 33.5786 14.7497 35.3555L2.42346 58.7569C0.366937 62.6611 5.21921 66.5303 8.56771 63.6563Z'
              strokeWidth='2'
            />
          </svg>
        </button>
      )}
    </div>
  );
};

export default Carousel;
