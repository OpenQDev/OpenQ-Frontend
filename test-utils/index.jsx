// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
