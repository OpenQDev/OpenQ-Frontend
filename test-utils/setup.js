import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
