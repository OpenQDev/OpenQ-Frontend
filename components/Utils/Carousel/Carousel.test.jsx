/**
 * @vi-environment jsdom
 */
import React from 'react';
import { render, screen, waitFor } from '../../../test-utils';
import Carousel from '.';
import CarouselBounty from '../../Bounty/CarouselBounty';
import nextRouter from 'next/router';
import ShallowRenderer from 'react-test-renderer/shallow';
// Test cases for full balances, empty balances, and undefined balances.

nextRouter.useRouter = vi.fn();
const push = vi.fn(() => {
  return { catch: vi.fn };
});
const bounty = {
  id: 'I_kwDOGAqhQc48U5_r',
  title: 'test2',
  body: 'body of test2',
  url: 'https://github.com/OpenQDev/OpenQ-Frontend/issues/8',
  languages: [
    {
      __typename: 'Language',
      name: 'JavaScript',
      color: '#f1e05a',
    },
    { __typename: 'Language', name: 'CSS', color: '#563d7c' },
    { __typename: 'Language', name: 'Dockerfile', color: '#384d54' },
    { __typename: 'Language', name: 'Shell', color: '#89e051' },
  ],
  repoName: 'OpenQ-Frontend',
  owner: 'OpenQDev',
  avatarUrl: 'https://avatars.githubusercontent.com/u/77402538?v=4',
  labels: [
    {
      __typename: 'Label',
      name: 'documentation',
      color: '0075ca',
    },
    {
      __typename: 'Label',
      name: 'duplicate',
      color: 'cfd3d7',
    },
    {
      __typename: 'Label',
      name: 'enhancement',
      color: 'a2eeef',
    },
    {
      __typename: 'Label',
      name: 'wontfix',
      color: 'ffffff',
    },
    {
      __typename: 'Label',
      name: 'javascript',
      color: 'E16391',
    },
    {
      __typename: 'Label',
      name: 'bounty',
      color: '727384',
    },
  ],
  createdAt: '1661767948472',
  closed: true,
  bodyHTML: '<p dir="auto">body of test2</p>',
  titleHTML: 'test2',
  assignees: [],
  number: 8,
  repoUrl: 'https://github.com/OpenQDev/OpenQ-Frontend',
  repoDescription: null,
  prs: [],
  __typename: 'Bounty',
  bountyAddress: '0x3c57cd59335d9a56b76feb48b3925d095e7e8e9f',
  bountyId: 'I_kwDOGAqhQc48U5_r',
  bountyMintTime: '1661767971',
  bountyClosedTime: null,
  status: '0',
  fundingGoalTokenAddress: '0x0000000000000000000000000000000000000000',
  fundingGoalVolume: '0',
  payoutTokenVolume: '100',
  payoutTokenAddress: '0x5fbdb2315678afecb367f032d93f642f64180aa3',
  payoutSchedule: null,
  closerData: null,
  bountyType: '1',
  claimedTransactionHash: null,
  deposits: [
    {
      __typename: 'Deposit',
      id: '0x5fad7d4850474383e594afa53cedd57709f994d3da7787d72ecd4d4f0b6e1264',
      refunded: false,
      refundTime: null,
      expiration: '30',
      tokenAddress: '0x5fbdb2315678afecb367f032d93f642f64180aa3',
      volume: '2000000000000000000',
      sender: { __typename: 'User', id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266' },
      receiveTime: '1661768003',
    },
    {
      __typename: 'Deposit',
      id: '0xc468cf93bc351051c605130e29dd256ed200bd2d42f598d5dc1cb2b19ddd1743',
      refunded: false,
      refundTime: null,
      expiration: '30',
      tokenAddress: '0x0000000000000000000000000000000000000000',
      volume: '1000000000000000000',
      sender: { __typename: 'User', id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266' },
      receiveTime: '1661768000',
    },
    {
      __typename: 'Deposit',
      id: '0xc584d8d700359e89dbae47c6e6bf293d21ce250b67cc6b679636d0453cb86884',
      refunded: false,
      refundTime: null,
      expiration: '1296000',
      tokenAddress: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
      volume: '2000000000000000000',
      sender: { __typename: 'User', id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266' },
      receiveTime: '1661768002',
    },
    {
      __typename: 'Deposit',
      id: '0xf01aecc05f1e304ffaacb3ed1af91d23fd4be565e7b376660efa2916cf657f44',
      refunded: false,
      refundTime: null,
      expiration: '30',
      tokenAddress: '0x0000000000000000000000000000000000000000',
      volume: '1000000000000000000',
      sender: { __typename: 'User', id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266' },
      receiveTime: '1661768001',
    },
  ],
  refunds: [],
  payouts: [],
  issuer: { __typename: 'User', id: '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266' },
  bountyTokenBalances: [
    {
      __typename: 'BountyFundedTokenBalance',
      volume: '2000000000000000000',
      tokenAddress: '0x0000000000000000000000000000000000000000',
    },
    {
      __typename: 'BountyFundedTokenBalance',
      volume: '2000000000000000000',
      tokenAddress: '0x5fbdb2315678afecb367f032d93f642f64180aa3',
    },
    {
      __typename: 'BountyFundedTokenBalance',
      volume: '2000000000000000000',
      tokenAddress: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
    },
  ],
  tvl: 0,
  address: '0x3C57cD59335D9a56b76fEB48b3925D095e7e8E9f',
  organizationId: 'MDEyOk9yZ2FuaXphdGlvbjc3NDAyNTM4',
  type: '1',
  category: null,
  watchingCount: 0,
  blacklisted: false,
  organization: { blacklisted: false, __typename: 'Organization' },
};
describe('Carousel', () => {
  // Test cases for

  beforeEach(() => {
    nextRouter.useRouter.mockImplementation(() => ({
      query: { type: null },
      prefetch: vi.fn(() => {
        return { catch: vi.fn };
      }),
      push,
    }));
  });
  it('should match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<Carousel height={36}></Carousel>);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
  it('should display repo name and title', async () => {
    // ARRANGE

    render(
      <Carousel height={36}>
        <CarouselBounty bounty={bounty} />
        <CarouselBounty bounty={bounty} />
        <CarouselBounty bounty={bounty} />
        <CarouselBounty bounty={bounty} />

        <CarouselBounty bounty={bounty} />
      </Carousel>
    );
    // ASSERT
    await waitFor(async () => {
      expect(screen.getAllByText(/OpenQDev\/openq-frontend/i)).toHaveLength(5);
      expect(await screen.findAllByText(/test2/i)).toHaveLength(5);
    });
  });
});
