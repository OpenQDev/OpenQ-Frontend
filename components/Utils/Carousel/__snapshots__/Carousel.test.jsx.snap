// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > should match DOM Snapshot 1`] = `
<div
  className="relative undefined justify-center self-center w-full flex undefined"
  onMouseUp={[Function]}
>
  <div
    className="flex gap-2 xl:max-w-[100%] xl:w-[980px] undefined overflow-x-auto justify-items-center"
    data-testid="carousel"
  />
</div>
`;
